src/test/java/org/eclipse/basyx/testsuite/regression/submodel/restapi/SubmodelProviderTest.java

```diff
@@ -391,6 +394,114 @@ public void testUpdateElementInSmElementCollection() {
 		assertEquals(321, value.intValue());
 	}
 
+	@SuppressWarnings("unchecked")
+	@Test
+	public void collectionIdShortFileCollision() {
+		VABElementProxy submodelProxy = getConnectionManager().connectToVABElement(submodelAddr);
+
+		String colIdShort = "Files";
+		String fileIdShort = "File";
+
+		SubmodelElementCollection fileCollection = new SubmodelElementCollection(colIdShort);
+		File fileSubmodelElement = new File("application/xml");
+		String fileValue = "simpleFile.xml";
+		fileSubmodelElement.setValue(fileValue);
+		fileSubmodelElement.setIdShort(fileIdShort);
+		fileCollection.addSubmodelElement(fileSubmodelElement);
+
+		String smeCollectionPath = VABPathTools.concatenatePaths(SMPROVIDER_PATH_PREFIX, MultiSubmodelElementProvider.ELEMENTS, colIdShort);
+		submodelProxy.setValue(smeCollectionPath, fileCollection);
+
+		String fileValuePath = VABPathTools.concatenatePaths(smeCollectionPath, fileIdShort);
+		Map<String, Object> resultingFileMap = (Map<String, Object>) submodelProxy.getValue(fileValuePath);
+		File resultingFile = File.createAsFacade(resultingFileMap);
+		assertEquals(fileValue, resultingFile.getValue());
+	}
+
+	/**
```

Comment:
Please remove this comment by refactoring the function name - additionally, the created file has not "File" as IdShort but "MySimpleFile". Is this an error?

Reply:
The comment needs to be moved down one method. The ``getIdShortCollision`` method is what is referred to.